controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -64,6 +64,7 @@ type MachineGenerator interface {
 // +kubebuilder:rbac:groups=core,resources=events,verbs=get;list;watch;create;patch
 // +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;patch
 // +kubebuilder:rbac:groups=controlplane.cluster.x-k8s.io,resources=kubeadmcontrolplanes;kubeadmcontrolplanes/status,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;machines;machinesets;machinedeployments;machinepools,verbs=get;list;watch
```

Comment:
I wouldn't expect the control plane to need MachineSets, MachineDeployments, or MachinePools

Reply:
let me see, i was getting errors when it was trying to start up, but i also had a cluster role name mismatch, so fixing that might have cleared these up, one sec

Reply:
yeah works fine without, updating

Reply:
never mind, it does not work without these RBAC rules, this is a necessary change.

Reply:
I can see the need for Clusters and Machines, but the others shouldn't be needed.

Reply:
I'll look a little deeper to see if there's something weird going on. For now, take a look at these logs: https://cloud.tilt.dev/snapshot/Afyl8-ALE9PJ15YiKds=

Reply:
poo, looks like the snapshot feature is not working :(